neuralpp/experiments/experimental_inference/belief_propagation.py

```diff
@@ -0,0 +1,29 @@
+from typing import List, Optional
+
+from neuralpp.inference.graphical_model.representation.factor.product_factor import (
+    Factor, ProductFactor
+)
+from neuralpp.inference.graphical_model.variable.variable import Variable
+
+
+class BeliefPropagation:
+
+    @staticmethod
+    def run(query: Variable, factors: List[Factor]):
+        return BeliefPropagation.msg_from_variable(query, None, factors).normalize()
+
+    @staticmethod
+    def msg_from_factor(factor: Factor, out_var: Variable, factors: List[Factor]):
+        neighbors = [v for v in factor.variables if v != out_var]
+        if not neighbors:
```

Comment:
I've just created a static method `ProductFactors.multiply(factors)` which returns `ProductFactors(factors)` if there are 0 or more than 1 factors in the list, and `factor[0]` if `len(factors) == 1`. If you update your repo, you can use it here and eliminate the `if not neighbors`.